File: my-folder/0013-roman-to-integer/solution.c
int romanToInt(char* s) {
    int map[256] = {0};
    map['I'] = 1;
    map['V'] = 5;
    map['X'] = 10;
    map['L'] = 50;
    map['C'] = 100;
    map['D'] = 500;
    map['M'] = 1000;

    int result = 0;
    int prev = 0;

    while (*s) {
        int current = map[(unsigned char)*s];
        result += current;
        if (prev < current) {
            result -= 2 * prev;
        }
        prev = current;
        s++;
    }

    return result;
}

File: my-folder/0013-roman-to-integer/solution.rs
impl Solution {
    pub fn roman_to_int(s: String) -> i32 {

        let mut map = std::collections::HashMap::new();
        map.insert('I', 1);
        map.insert('V', 5);
        map.insert('X', 10);
        map.insert('L', 50);
        map.insert('C', 100);
        map.insert('D', 500);
        map.insert('M', 1000);

        let mut result = 0;
        let mut prev = 0;

        for c in s.chars() {
            let current = *map.get(&c).unwrap_or(&0);
            result += current;
            if prev < current {
                result -= 2 * prev;
            }
            prev = current;
        }

        result
    }
}
